ruma-federation-api/src/user/get_devices/v1.rs

```diff
@@ -0,0 +1,35 @@
+//! [GET /_matrix/federation/v1/user/devices/](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-user-devices-userid)
+
+use ruma_api::ruma_api;
+use ruma_client_api::r0::Device;
+use ruma_identifiers::{DeviceId, UserId};
+
+ruma_api! {
+    metadata: {
+        description: "Gets information on all of the user's devices.",
+        name: "get_devices",
+        method: GET,
+        path: "/_matrix/federation/v1/user/devices",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request: {
+        /// The user ID to retrieve devices for. Must be a user local to the receiving homeserver.
+        #[ruma_api(query)]
+        pub user_id: UserId,
```

Comment:
This isn't documented and applied to all other endpoints yet, but request fields should generally be `Copy` now. You can just pretend a named lifetime is in scope and make this a reference; the macro will take care of the rest:

```suggestion
        pub user_id: &'a UserId,
```